loyalty_card_fixed_expiration_date/models/loyalty_card.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 (APSL-Nagarro) - Antoni Marroig
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from datetime import date, timedelta
+
+from odoo import api, models
+
+
+class LoyaltyCard(models.Model):
+    _inherit = "loyalty.card"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        # If the expiration date is not set, we set it to
+        # the current date + the fixed duration
+        for record, vals in zip(res, vals_list, strict=True):
+            if not vals.get("expiration_date", False):
+                if record.program_id and record.program_id.duration_days > 0:
+                    record.expiration_date = date.today() + timedelta(
```

Comment:
In order to allow inheritance, I would have done a method `_get_expiration_date()` and maybe a method to embed the whole code here in `_set_expiration_date()`